libs/datamodel/core/src/configuration/datasource.rs

```diff
@@ -32,13 +35,18 @@ impl std::fmt::Debug for Datasource {
             .field("url", &"<url>")
             .field("documentation", &self.documentation)
             .field("active_connector", &&"...")
-            .field("shadow_database_url", &self.shadow_database_url)
+            .field("shadow_database_url", &"<shadow_database_url>")
```

Comment:
Just me curious with 0 Rust knowledge: What is the difference?

Reply:
That should be a separate commit. This is the `Debug` representation of a datasource. Usually you use `#[derive(Debug)]` on a struct to implement that for you, but since we don't want to leak credentials, we have a manual implementation that hides the contents of the `url` property on the datasource. I noticed while working on the multi-schema feature that we were printing the shadow database url contents in there, but that's also potentially a credential leaks, so now instead of printing the shadow database url, it literally prints `<shadow_database_url>`. This should never be shown to users anyway, but since it might, by mistake, it felt safer to remove the credentials from there.